zookeeper-server/src/main/java/org/apache/zookeeper/jmx/ManagedUtil.java

```diff
@@ -69,9 +69,13 @@ public static void registerLog4jMBeans() throws JMException {
             try {
                 // Create and Register the top level Log4J MBean
                 // org.apache.log4j.jmx.HierarchyDynamicMBean hdm = new org.apache.log4j.jmx.HierarchyDynamicMBean();
-                Object hdm = Class.forName("org.apache.log4j.jmx.HierarchyDynamicMBean").getDeclaredConstructor().newInstance();
+                Object hdm = Class.forName("org.apache.log4j.jmx.HierarchyDynamicMBean").getConstructor().newInstance();
 
-                ObjectName mbo = new ObjectName("log4j:hiearchy=default");
+                String mbean = System.getenv("zookeeper.jmx.log4j.mbean");
```

Comment:
You should use System.getProperty(name,defaultvalue)

Reply:
> I would go for a system property

FYI, `System.getenv()` is for Operating System environment variables, whereas `System.getProperty()` is for JVM arguments which are passed as `-DpropName=value` to Java application launcher.

So, maybe use `getenv()` would be more reasonable.

Reply:
As far as I know in ZK it is common to use System Properties and not environment variables.

I think that the launcher for the server process is setting as system property every configuration parameter.

Do you have any example of usages of getEnv in ZooKeeper ?

Reply:
@eolivelli Okay, I will modify it.